src/components/ImportForm/utils/useValidApplicationName.ts

```diff
@@ -1,31 +1,33 @@
 import * as React from 'react';
-import { useK8sWatchResource } from '@openshift/dynamic-plugin-sdk-utils';
-import { ApplicationGroupVersionKind } from '../../../models';
+import { useApplications } from '../../../hooks/useApplications';
 import { ApplicationKind } from '../../../types';
 import { useWorkspaceInfo } from '../../../utils/workspace-context-utils';
 
 const BASE_NAME = 'my-app';
 
+export const incrementNameCount = (
+  applications: ApplicationKind[],
+  prefferedName?: string,
+): string => {
+  let name = prefferedName || BASE_NAME,
+    i = 1;
+  while (applications.some((app) => app.metadata.name === name)) {
+    name = `${prefferedName || BASE_NAME}-${i}`;
+    i++;
+  }
+  return name;
+};
+
 export const useValidApplicationName = (): [string, boolean, unknown] => {
```

Comment:
You would also need to remove the use of this hook from `ImportForm` where it is sending `recommendedApplicationName`.